### PR TITLE
Page Select and Page Turn

### PR DESCRIPTION
**Pull request by thehabes**
image

### CODE REVIEW

**Comment by thehabes** [comment]
image

File: css/gloss.css [hunk 1]
```diff
@@ -66,11 +66,16 @@ a[data-change] {
...
 }
 
+#folioLayout{
+  box-shadow: 0 0 .25rem #000000;
+  height: 100vh;
+}
+
 .page {
   display   : block;
+  height    : 98vh;
```
Comment (cubap) [inline css/gloss.css:76]: this can get nuts without a max set, but that's future problem

File: css/gloss.css [hunk 2]
```diff
@@ -223,12 +228,12 @@ line.selected.just {
 }
 
 #zoomy {
-  position        : fixed;
+  position        : relative;
```
Comment (cubap) [inline css/gloss.css:231]: makes sense

File: gloss.html [hunk 3]
```diff
@@ -15,13 +15,11 @@
             <a href="./index.html">🏠</a>
             <a href="./manuscripts.html">📚 View Manuscripts</a>
             <a href="./named-glosses.html">📑 Named Glosses</a>
-
             <a href="./ms.html" id="thisMS">📖 This Manuscript</a>
+            <auth-button class="container">
+                <button is="auth-button">login</button>
```
Comment (cubap) [inline gloss.html:20]: good call

File: js/auth.js [hunk 4]
```diff
@@ -49,6 +49,7 @@ class AuthButton extends HTMLButtonElement {
         this.onclick = logout
         this.login = login
         this.logout = logout
+        this.style.marginTop = "9px"
```
Comment (cubap) [inline js/auth.js:52]: weird.  I hate it.

File: js/deer-render.js [hunk 5]
```diff
@@ -182,7 +182,23 @@ DEER.TEMPLATES.thumbs = function (obj, options = {}) {
 }
 
 DEER.TEMPLATES.pageLinks = function (obj, options = {}) {
-    return obj.sequences[0].canvases.reduce((a, b, i) => a += `<a class="button" href="?page=${i + 1}#${obj["@id"]}">${b.label}</a>`, ``)
+    let params = (new URL(document.location)).searchParams
+    let highlight = parseInt(params.get("page")) || 1
+    const optionelems = obj.sequences[0].canvases.reduce((a, b, i) => a += `<option class="button" value="${i+1}" ${(highlight === i+1) ? "selected" : ""}>${b.label}</option>`, ``)
+    const select = `<select onchange="document.location=document.location.pathname+'?page='+event.target.value+'#${obj["@id"]}'">${optionelems}</select>`
```
Comment (cubap) [inline js/deer-render.js:188]: nicer

File: js/deer-render.js [hunk 6]
```diff
@@ -182,7 +182,23 @@ DEER.TEMPLATES.thumbs = function (obj, options = {}) {
...
+
+DEER.TEMPLATES.pageTurn = function (obj, options = {}) {
+    let params = (new URL(document.location)).searchParams
+    let page = parseInt(params.get("page")) || 1
+    let numPages = obj.sequences[0].canvases.length
+    let nextpage = (page === numPages) ? 0 : page + 1
+    let prevpage = (page === 1) ? 0 : page - 1
+    return`
+        <a class="button secondary ${(page === 1) ? "is-hidden" : ""}"  href="?page=${prevpage}#${obj["@id"]}">Previous Page</a>
+        <a class="button secondary pull-right ${(page === numPages) ? "is-hidden" : ""}" href="?page=${nextpage}#${obj["@id"]}">Next Page</a>
```
Comment (cubap) [inline js/deer-render.js:200]: isn't that what I said ;) ?

File: layout.html [hunk 7]
```diff
@@ -15,15 +15,22 @@
...
     </gm-header>
     <h3 id="glossTitle" class="deer-view send-id" deer-id="" deer-template="label">Layout Editor</h3>
     <input is="auth-creator" type="hidden" deer-key="creator" />
 
+    <div class="row">
+        <deer-view id="pageLinks" deer-id="" class="send-id col" deer-template="pageLinks"></deer-view>
+    </div>
+
+    <div class="row">
+        <deer-view id="pageTurn" deer-id="" class="send-id col" deer-template="pageTurn"></deer-view>
```
Comment (cubap) [inline layout.html:31]: this may get crazy with projects with lots of pages, but none that we have now

**Review by cubap — APPROVED** [review]
`creator` should be set from authenticated user, but otherwise, we're fine.

**Pull request merged**